packml_ros/include/packml_ros/packml_ros.h

```diff
@@ -44,24 +44,30 @@ class PackmlRos
   std::shared_ptr<packml_sm::AbstractStateMachine> sm_;
   ros::Publisher status_pub_;
   ros::Publisher stats_pub_;
+  ros::Publisher stats_transaction_pub_;
   ros::ServiceServer trans_server_;
   ros::ServiceServer reset_stats_server_;
   ros::ServiceServer get_stats_server_;
   ros::ServiceServer load_stats_server_;
   packml_msgs::Status status_msg_;
-  float stats_publish_period_;
+  double stats_publish_period_;
+  double stats_transaction_publish_period_;
   ros::Timer stats_timer_;
+  ros::Timer stats_transaction_timer_;
 
   bool transRequest(packml_msgs::Transition::Request& req, packml_msgs::Transition::Response& res);
 
 private:
   void handleStateChanged(packml_sm::AbstractStateMachine& state_machine, const packml_sm::StateChangedEventArgs& args);
   void getCurrentStats(packml_msgs::Stats& out_stats);
+  void getStatsTransaction(packml_msgs::Stats& out_stats);
+  packml_msgs::Stats populateStatsMsg(const packml_sm::PackmlStatsSnapshot& stats_snapshot);
   bool getStats(packml_msgs::GetStats::Request& req, packml_msgs::GetStats::Response& response);
   bool resetStats(packml_msgs::ResetStats::Request& req, packml_msgs::ResetStats::Response& response);
   bool loadStats(packml_msgs::LoadStats::Request& req, packml_msgs::LoadStats::Response& response);
   packml_sm::PackmlStatsSnapshot populateStatsSnapshot(const packml_msgs::Stats& msg);
   void publishStatsCb(const ros::TimerEvent& timer_event);
+  void publishStatsTransactionCb(const ros::TimerEvent &timer_event);
```

Comment:
Transaction might not be the best name for this function. Transactions imply multiple steps that either succeed together or are rolled back to prevent corruption of a record.

This functionality is intended to publish a difference (what has happened since the last time this was called) or incremental update.

At the very least, add some doxygen-style comments to clarify what the function seeks to accomplish, but it might be worth the work to refactor the naming here.

Reply:
Yeah, I can see this. So I renamed everything to incremental and added come comments, hopefully that will make this more clear.